### PR TITLE
Adds CIFAR tutorial + resnet

### PR DESCRIPTION
**Pull request by jdunnmon**
- [x] Adds CIFAR10 tutorial
- [x] Adds resnet class specifically for CIFAR10 tutorial
- [x] Uses updated API

Note that `resnet.py` is in `tutorials` not `contrib` because it is not actually a metal module -- it's just a generic encoder that we're using.   This CIFAR tutorial is meant to show how metal can be used to train a generic supervised model.

### CODE REVIEW

File: tutorials/resnet.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
+"""ResNet in PyTorch.
```
Comment (ajratner) [inline tutorials/resnet.py]: Put this into a new `contrib/modules` directory

File: tutorials/CIFAR_Tutorial.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
+import argparse
```
Comment (ajratner) [inline tutorials/CIFAR_Tutorial.py]: Can you add a brief blurb describing this tutorial (a) as a header comment in this file, and (b) in `tutorials/README.md`?

File: tutorials/CIFAR_Tutorial.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+from metal import EndModel
+from metal.utils import convert_labels
+
+SEED = 1
+
+# CUDA?
+cuda = torch.cuda.is_available()
+
+# For reproducibility
+torch.manual_seed(SEED)
```
Comment (ajratner) [inline tutorials/CIFAR_Tutorial.py]: Isn't this handled in the `EndModel` class?  Should confirm and if so just let be handled there

File: tutorials/CIFAR_Tutorial.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+if cuda:
+    torch.cuda.manual_seed(SEED)
+
+
+parser = argparse.ArgumentParser(description="PyTorch Training")
```
Comment (ajratner) [inline tutorials/CIFAR_Tutorial.py]: This is a tutorial so being slightly more verbose with explanatory comments is good- add one here saying e.g. 'setting up command line argument parser'

File: tutorials/CIFAR_Tutorial.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+        train_data=train_loader,
+        dev_data=test_loader,
+        l2=args.weight_decay,
+        lr=args.lr,
+        n_epochs=args.epochs,
+        print_every=1,
+        validation_metric="accuracy",
+    )
+
+    # Test end model
```
Comment (ajratner) [inline tutorials/CIFAR_Tutorial.py]: Should have some record of what score it should achieve- at least in the header comment at top (e.g. 'we should be able to reach XX% accuracy, although in this tutorial as configured, we don't train for as long so we get YY%. To get XX%, you would just...').  Possibly also add an assert statement here
